Sources/SwiftUINavigationCore/NavigationLink+Vanilla.swift

```diff
@@ -0,0 +1,55 @@
+#if canImport(SwiftUI)
+  import SwiftUI
+
+  @available(
+    iOS, introduced: 13, deprecated: 16,
+    message:
+      "use NavigationLink(value:label:), or navigationDestination(isPresented:destination:), inside a NavigationStack or NavigationSplitView"
+  )
+  @available(
+    macOS, introduced: 10.15, deprecated: 13,
+    message:
+      "use NavigationLink(value:label:), or navigationDestination(isPresented:destination:), inside a NavigationStack or NavigationSplitView")
+  @available(
+    tvOS, introduced: 13, deprecated: 16,
+    message:
+      "use NavigationLink(value:label:), or navigationDestination(isPresented:destination:), inside a NavigationStack or NavigationSplitView")
+  @available(
+    watchOS, introduced: 6, deprecated: 9,
+    message:
+      "use NavigationLink(value:label:), or navigationDestination(isPresented:destination:), inside a NavigationStack or NavigationSplitView"
+  )
+  extension NavigationLink {
+    /// Creates a navigation link that presents the view corresponding to a value.
+    ///
+    /// > Tip: This initializer is provided for applications deploying to older OSes that want to
+    /// > drive navigation with an optional.
+    ///
+    /// - Parameters:
+    ///   - value: A value to present.
+    ///   - item: A binding to the presented value.
+    ///   - destination: A view for the navigation link to present.
+    ///   - label: A view builder to produce a label describing the destination to present.
+    public init<T, D: View>(
+      value: @autoclosure @escaping () -> T,
+      item: Binding<T?>,
+      @ViewBuilder destination: (T) -> D,
+      @ViewBuilder label: () -> Label
+    ) where Destination == D? {
```

Comment:
I am now 100% on this API...

I've required a value up front since tapping eagerly navigates a link. Technically a person throwing this in a background could work around the issue with a closure to a fatal error:

```swift
.background {
  NavigationLink(
    value: { fatalError() },
    item: $store.scope(state: \.child, action: \.child)
  ) { store in
    ChildView(store: store)
  } label: {}
}
```

We _could_ backport `navigationDestination` even _further_ driven by a hidden `NavigationLink`...but again we probably don't want to be in the business of providing too many SwiftUI tools that stray from the actual underlying tools.

Reply:
And a hidden `NavigationLink` doesn't work unless it's on the screen correct?

Reply:
@agarrharr Yep, one of the many gotchas and probably another reason why we don't want to ship an API that is confusing because of the underlying API it calls to :smile: